### ISSUE TITLE
[BUG] Anycast Clusterer, sending replies directly.

### ISSUE THREAD

**Issue by digipigeon** [OpenSIPS/opensips#2445]
**OpenSIPS version you are running**

```
version: opensips 3.0.5 (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, USE_MCAST, SHM_MMAP, PKG_MALLOC, Q_MALLOC, F_MALLOC, HP_MALLOC, DBG_MALLOC, FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16, MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll, sigio_rt, select.
git revision: b7b1518e9
main.c compiled on  with gcc 7
```

**Describe the bug**

If the an Anycast node receives a fail reply belonging to another node in the cluster, as well as passing the message back to the responsible node a message is send to the originator directly as well.

**To Reproduce**

1. Setup Anycast cluster as a dispatcher
2. Point Anycast cluster to a setup a system to return a fail with a 500 failure.
3. Send INVITE to a node non-local to the system which returns a fail.
4. The Anycast node local to the machine with fails with 500 propagates the message incorrectly.

**Expected behavior**

The 500 error message should not propagate to the originator, only pass internally to the responsible node in the cluster.

**Relevant System Logs**

See Attached
[overloaded-direct.txt](https://github.com/OpenSIPS/opensips/files/6206276/overloaded-direct.txt)

**OS/environment information**
 - Operating System:  Ubuntu 18.04
 - OpenSIPS installation: apt repo, latest nightly

**Comment by razvancrainea** [comment]
Just pushed a fix in the master branch. Any way you could test it in your enviroment?

**Comment by razvancrainea** [comment]
hey, @digipigeon ! Any chance you could give it a try?

**Comment by digipigeon** [comment]
Sure, I'll give this a go shortly.

**Comment by razvancrainea** [comment]
@digipigeon any chance you could give it a try before tomorrow's freeze? Thanks!

**Comment by razvancrainea** [comment]
As I'm pretty confident in the fix, I've backported it in 2.4. However, it would be nice if you could give it a try as well before the minor release, just to have a second pair of eyes.

**Comment by digipigeon** [comment]
Yes, this seems to have fixed this issue, thank you.
